openmls/src/framing/mls_auth_content.rs

```diff
@@ -281,16 +280,35 @@ impl AuthenticatedContent {
     pub fn sender(&self) -> &Sender {
         &self.content.sender
     }
+}
 
-    #[cfg(test)]
+#[cfg(test)]
+impl AuthenticatedContent {
     pub fn test_signature(&self) -> &Signature {
         &self.auth.signature
     }
 
-    #[cfg(test)]
     pub(super) fn unset_confirmation_tag(&mut self) {
         self.auth.confirmation_tag = None;
     }
+
+    pub(crate) fn from_bytes(bytes: &[u8]) -> Result<Self, tls_codec::Error> {
```

Comment:
Only this is new.

Reply:
Why is this needed? Can't you just derive this for test builds?

Reply:
`derive` is not possible due to `FramedContentAuthData::deserialize` that requires the content type. But yes, I should already do this properly: I will change this to a test-gated custom tls_codec implementation and do the trailing data check in the test for now.

We should think about providing a blanket implementation that provides `from_bytes` for everything that is `tls_codec::deserialize`. Sometimes we want to ensure there is no trailing data after deserialize and this would be a very effective way to get that for everything. What do you think?